### ISSUE TITLE
Writing Text in Language with Right to Left Direction

### ISSUE THREAD

**Issue by msrosyidi** [Bramas/texiteasy#12]
I often write using at least two languages with different writing direction, one has Right to Left direction (arabic, hebrew) and other has Left to Right direction (english, indonesian). TexWorks and TexStudio both have full support for writing mixed languages with different writing direction, but the first has not syntax completion and the second is very-very heavy so I only use both rarely. Usually I use Notepad++ for editing but it also has a bad support for writing mixed languages with different writing direction.

1. TexWorks
![texworks](https://cloud.githubusercontent.com/assets/3489885/10411191/721cd43c-6f87-11e5-8130-3ba40cf211bf.png)
2. TexStudio
![texstudio](https://cloud.githubusercontent.com/assets/3489885/10411193/8c0b454a-6f87-11e5-9442-8159d6173814.png)
3. Notepad++
In LTR View, all texts are displayed in LTR direction
![notepad ltrview](https://cloud.githubusercontent.com/assets/3489885/10411289/fe447792-6f89-11e5-8d22-b0119a359b90.png)
In RTL View all texts are displayed in RTL direction
![notepad rtlview](https://cloud.githubusercontent.com/assets/3489885/10411296/1d6ce172-6f8a-11e5-97bf-5c138e249cd4.png)
In my opinion It is so bad and I use it for tex editing just because it can run fast and has syntax completion for latex. But for tex compilation I need to manually add some custom setting and command to it using NPPExec.
4. TexitEasy
![texiteasy](https://cloud.githubusercontent.com/assets/3489885/10411200/b40ea708-6f87-11e5-9784-7712f10247ae.png)
The text in arabic and hebrew are difficult to write and edited when the editor does not display in right to left direction and Arabic text is more difficult when I need to write the vocal marks.

Do you have any plans to give RTL support to this interesting tex editor in the future release? I'm very interesting to use TexiEasy, it runs fast and lightly in my computer but does not lose many features needed for tex editing (syntax completion, tools for tex compilation, etc). Only one thing I lose it from this editor, an RTL support.

Thanks and I'm sorry for my bad english.

**Comment by Bramas** [comment]
Hi thanks for the feedback and for the precise explanation.
RTL support of texiteasy is directly linked to the Qt Library it uses. I think Qt worked hard to support RTL editing but, as you pointed out, the layout alignment is not RTL.

I tried at home and, if I understand well, the RTL editing is supported but the alignement is not correct. This may come from the fact that Qt allows multiple langage types in the same line so it does not know which alignement to apply.

It will be hard for me to change how Qt handle RTL langage, but (if the alignment is the main problem) maybe I can add a feature to choose the alignement of a block (and all new block obtained from this block). Maybe a link in a menu to say that the current line is RTL (or LTR) so that it change the alignement. Then if we press Enter to create a new line, the next line will have the same alignement. Do you think this could do the job?

**Comment by msrosyidi** [comment]
Yes, I think that is what I need or near to it. So when I want to give the vowel mark I can write it easily. It is a suggested way in writing arabic to write the entire consonant and write the vocal mark as the last step, and it is so hard if the alignment of the text is not right to left. Thanks for your support.

**Comment by msrosyidi** [comment]
Oh yes, I forget to mention yesterday that when I look at the installation directory of TeXstudio I see there are many dll files named with prefix Qt5. I'm not sure TeXstudio is built using QTLibrary or not, I do not understand the C language, but as I mentioned above that only TexWorks and TexStudio have full support for writing mixed languages with different writing direction as I need.

**Comment by Bramas** [comment]
Texstudio use Qt5, but not for the editor. The editor is an indépendant project called QCodeEdit. That's the reason of the different features, and the heaviness as well.

**Comment by msrosyidi** [comment]
Thanks for the explanation.

**Comment by renanpoco** [comment]
Can you fix an output bug ? i mean, my input texts are written from the right to left like hebrew/persian/farsi/etc and my xelatex script is generating images of those texts in the wrong order (from left to the right).
For example

input: אבא שלי 
output: שלי אנא

I tried to change the line :

self.make_string_xelatex_safe(text),

to

self.make_string_xelatex_safe(text[::-1]),

but it reversed all the letters instead of words position only.

Thanks,
